### ISSUE TITLE
Failing on ECMWF ensemble

### ISSUE THREAD

**Issue by guidocioni** [gribscan/gribscan#18]
I was just giving this a try. 

I have 3 files coming from the ECMWF ensemble with all the members inside. 

<img width="710" alt="Screen Shot 2022-09-02 at 09 16 32" src="https://user-images.githubusercontent.com/12760310/188080409-b4b9c77a-2f9d-4301-bb6a-a713da4da368.png">

I just tried to 

```bash
gribscan-index *.grib2 -n 8
gribscan-build *.index -o dataset.json --prefix /home/ekman/ssd/guido/ecmwf-ens/
```

But received 

```python
Traceback (most recent call last):
  File "/home/ekman/miniconda3/envs/models/bin/gribscan-build", line 8, in <module>
    sys.exit(build_dataset())
  File "/home/ekman/miniconda3/envs/models/lib/python3.10/site-packages/gribscan/tools.py", line 29, in build_dataset
    refs = gribscan.grib_magic(args.indices, global_prefix=args.prefix)
  File "/home/ekman/miniconda3/envs/models/lib/python3.10/site-packages/gribscan/gribscan.py", line 424, in grib_magic
    global_attrs, coords, varinfo = inspect_grib_indices(messages, magician)
  File "/home/ekman/miniconda3/envs/models/lib/python3.10/site-packages/gribscan/gribscan.py", line 323, in inspect_grib_indices
    dims, dim_id, shape = map(tuple, zip(*((dim, i, len(coords))
ValueError: not enough values to unpack (expected 3, got 0)
```

I don't know whether it's because of the files dimensions or whether I'm doing something wrong.

**Comment by d70-t** [comment]
I've seen something like this before. I think this was related to variables which occur only in a single message (e.g. a single non-time-varying horizontal field).  https://github.com/gribscan/gribscan/commit/2899eac1675acb5ad3f7d64e23103c485c806956 is probably what you need. It's included in the current `main`, but unfortunately the latest PyPi release is one commit earlier 😬.

* @guidocioni can you try with the latest `main` version?
* @lkluft do you think we could get some new release?

**Comment by guidocioni** [comment]
> I've seen something like this before. I think this was related to variables which occur only in a single message (e.g. a single non-time-varying horizontal field). [2899eac](https://github.com/gribscan/gribscan/commit/2899eac1675acb5ad3f7d64e23103c485c806956) is probably what you need. It's included in the current `main`, but unfortunately the latest PyPi release is one commit earlier 😬.
> 
> * @guidocioni can you try with the latest `main` version?
> * @lkluft do you think we could get some new release?

Let me try again and I'll get back to you.

**Comment by guidocioni** [comment]
Hey, so with the correction that you posted it seems to work, albeit the first step being slower.

For the second step I had to manually create a folder `dataset.json`, because otherwise it was complaining 

```python
Traceback (most recent call last):
  File "/home/ekman/miniconda3/envs/models/bin/gribscan-build", line 8, in <module>
    sys.exit(build_dataset())
  File "/home/ekman/miniconda3/envs/models/lib/python3.10/site-packages/gribscan/tools.py", line 32, in build_dataset
    with open(f"{args.output}/{dataset}.json", "w") as indexfile:
FileNotFoundError: [Errno 2] No such file or directory: 'dataset.json/atm2d.json'
```

After creating the folder it wrote a `json` file `dataset.json/atm2d.json`.
I'm not sure if I'm doing things correctly because I imagine you would have already included a check to see if that folder is existing or not...so it smells like I'm using one of the parameters wrong.

Anyhoe...when I tried to load into xarray 

```python
ds = xr.open_zarr("reference::/home/ekman/ssd/guido/ecmwf-ens/dataset.json/atm2d.json", consolidated=False)
```

Then I get a dataset with collapsed dimensions. 
<img width="800" alt="Screenshot 2022-09-02 at 14 08 53" src="https://user-images.githubusercontent.com/12760310/188139911-5a8810ae-85fd-4bc1-b8de-847612acfca7.png">

Again, sorry if I'm doing something wrong: `zarr` is kind of new to me.

**Comment by trackow** [comment]
This looks exactly like what I would expect. I also normally create the folder -o dataset.json or -o json.dir myself before running the gribscan-build command.
